doc/source/data/training_ingest.rst

```diff
@@ -0,0 +1,525 @@
+
+To customize this, pass in a :class:`DataConfig <ray.train.DataConfig>` to the Trainer constructor. For example, to split both the training and validation datasets, do the following:
+
+.. testcode::
+    import ray
+    from ray.air import ScalingConfig, session
+    from ray.train.torch import TorchTrainer
+
+    ds = ray.data.read_text(
+        "s3://anonymous@ray-example-data/sms_spam_collection_subset.txt"
+    )
+    train_ds, val_ds = dataset.train_test_split(0.3)
+
+    def train_loop_per_worker():
+        # Get an iterator to the dataset we passed in below.
+        it = session.get_dataset_shard("train")
+        for _ in range(2):
+            for batch in it.iter_batches(batch_size=128):
+                print("Do some training on batch", batch)
+
+    my_trainer = TorchTrainer(
+        train_loop_per_worker,
+        scaling_config=ScalingConfig(num_workers=2),
+        datasets={"train": train_ds, "val": val_ds},
+        dataset_config=ray.train.DataConfig(
+            datasets_to_split=["train", "val"],
+        ),
+    )
+    my_trainer.fit()
+
+.. testoutput::
+    :hide:
+
+    ...
+
+Full customization (advanced)
+#############################
+For use cases not covered by the default config class, you can also fully customize exactly how your input datasets are splitted. To do this, you need to define a custom :class:`DataConfig <ray.train.DataConfig>` class (DeveloperAPI). The :class:`DataConfig <ray.train.DataConfig>` class is responsible for that shared setup and splitting of data across nodes.
+
+.. testcode::
+
+    # Note that this example class is doing the same thing as the basic DataConfig
+    # impl included with Ray Train.
+    from typing import Optional, Dict, List
+
+    import ray
+    from ray.air import ScalingConfig, session
+    from ray.train.torch import TorchTrainer
+    from ray.data import Dataset, DataIterator, NodeIdStr
+    from ray.actor import ActorHandle
+
+    ds = ray.data.read_text(
+        "s3://anonymous@ray-example-data/sms_spam_collection_subset.txt"
+    )
+
+    def train_loop_per_worker():
+        # Get an iterator to the dataset we passed in below.
+        it = session.get_dataset_shard("train")
+        for _ in range(2):
+            for batch in it.iter_batches(batch_size=128):
+                print("Do some training on batch", batch)
+
+
+    class MyCustomDataConfig(DataConfig):
+        def configure(
+            self,
+            datasets: Dict[str, Dataset],
+            world_size: int,
+            worker_handles: Optional[List[ActorHandle]],
+            worker_node_ids: Optional[List[NodeIdStr]],
+            **kwargs,
+        ) -> List[Dict[str, DataIterator]]:
+            assert len(datasets) == 1, "This example only handles the simple case"
+
+            # Configure Ray Data for ingest.
+            ctx = ray.data.DataContext.get_current()
```

Comment:
What's the scope of this get_current() context? Is this for every ray actor or a global singleton?

Reply:
global singleton